### PR TITLE
Bind a exporter to only needed tcp version

### PR DESCRIPTION
**Pull request by alexeymyltsev**
This PR bind exporter only to needed tcp version. 
If ip did not provided exporter will listen on all tcp versions.
It should solve https://github.com/openstack-exporter/openstack-exporter/issues/144

### CODE REVIEW

**Comment by niedbalski** [comment]
@alexeymyltsev Thank you for looking into this!. please rebase your patch and check the failing tests.

File: main.go [hunk 1]
```diff
@@ -63,7 +65,28 @@ func main() {
...
+	}
+
+	if enabledExporters == 0 {
+		log.Errorln("No exporter has been enabled, exiting")
+		os.Exit(-1)
+	}
+
+	sm := http.NewServeMux()
+	sm.Handle(*metrics, promhttp.Handler())
+	sm.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
```
Comment (niedbalski) [inline main.go:89]: why is this change required? lines 68 to 89

File: main.go [hunk 2]
```diff
@@ -84,8 +107,31 @@ func main() {
...
+	if err != nil {
+		log.Fatal(err)
+	}
+
 	log.Infoln("Starting HTTP server on", *bind)
-	log.Fatal(http.ListenAndServe(*bind, nil))
+	log.Fatal(http.Serve(l, sm))
+}
+
+func ip4or6(s string) string {
```
Comment (niedbalski) [inline main.go:121]: Would be possible to move this function into a utils.go file or module alongside with removeElements?

**Review by niedbalski — CHANGES_REQUESTED** [review]
Hello @alexeymyltsev thank you for this fix. left a minor set of nitpicks as code reviews.

File: main.go (hunk 1 above)
Comment (alexeymyltsev) [reply main.go]: You are right, i didn't dig enough to http lib.
Now it should be better.

File: main.go (hunk 2 above)
Comment (alexeymyltsev) [reply main.go]: done.
Please fix comment for exporters.RemoveElements if it is not clear description.

**Comment by niedbalski** [comment]
LGTM
